server/src/main/java/org/elasticsearch/cluster/service/MasterService.java

```diff
@@ -533,19 +582,13 @@ void notifyFailedTasks() {
                 assert executionResults.containsKey(updateTask.task) : "missing " + updateTask;
                 final ClusterStateTaskExecutor.TaskResult taskResult = executionResults.get(updateTask.task);
                 if (taskResult.isSuccess() == false) {
-                    updateTask.listener.onFailure(taskResult.getFailure());
+                    updateTask.onFailure(taskResult.getFailure());
                 }
             }
         }
 
         void notifySuccessfulTasksOnUnchangedClusterState() {
-            nonFailedTasks.forEach(task -> {
-                if (task.listener instanceof AckedClusterStateTaskListener) {
-                    // no need to wait for ack if nothing changed, the update can be counted as acknowledged
-                    ((AckedClusterStateTaskListener) task.listener).onAllNodesAcked(null);
-                }
-                task.listener.clusterStateProcessed(newClusterState, newClusterState);
-            });
+            nonFailedTasks.forEach(task -> { task.clusterStateUnchanged(newClusterState); });
```

Comment:
```suggestion
            nonFailedTasks.forEach(task -> task.clusterStateUnchanged(newClusterState));
```

Reply:
:+1: done in df25e1a66b86e7b4cf30cb1d5b3a6ecd65566770